src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,145 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            lastNode = new Node<>(null, value, null);
+            firstNode = lastNode;
+        } else {
+            Node<T> newElement = new Node<>(lastNode, value, null);
+            lastNode.next = newElement;
+            lastNode = newElement;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkBounds(index);
+        if (index == 0) {
+            Node<T> current = new Node<>(null, value, firstNode);
+            firstNode.previous = current;
+            firstNode = current;
+        } else {
+            Node<T> current = getNode(index);
+            Node<T> newNode = new Node<>(current.previous, value, current);
+            current.previous.next = newNode;
+            current.previous = newNode;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T element : list) {
+            add(element);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkBounds(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkBounds(index);
+        Node<T> current = getNode(index);
+        T oldValue = current.value;
+        current.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkBounds(index);
+        Node<T> removedElement = getNode(index);
+        final T result = removedElement.value;
+        if (removedElement.next == null) {
+            lastNode = removedElement.previous;
+        } else {
+            removedElement.next.previous = removedElement.previous;
+        }
+        if (removedElement.previous == null) {
+            firstNode = removedElement.next;
+        } else {
+            removedElement.previous.next = removedElement.next;
+        }
+        size--;
+        return result;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = firstNode;
+        for (int i = 0; i < size; i++) {
+            if (object == current.value
+                    || object != null && object.equals(current.value)) {
+                remove(i);
+                return true;
+            }
```

Comment:
It is not a good idea to call remove(i). Because you will have O(n2) complexity as in the remove by index you are using a loop again to find an item. Besides in the readme it is mentioned that you have to create unlink() method and call it from both methods remove(T object) and remove(int index)